DiscordChatExporter.Cli/Commands/Base/ExportCommandBase.cs

```diff
@@ -136,7 +136,7 @@ public string OutputPath
     private ChannelExporter? _channelExporter;
     protected ChannelExporter Exporter => _channelExporter ??= new ChannelExporter(Discord);
 
-    protected async ValueTask ExecuteAsync(IConsole console, IReadOnlyList<Channel> channels)
+    protected async ValueTask ExecuteAsync(IConsole console, IReadOnlyList<Channel> channels, IReadOnlyList<ChannelThread> threads)
```

Comment:
Can we have a unified export process that just takes `IReadOnlyList<IChannel>` instead? Then most of the code could be kept as is.

Reply:
I tried this, but I'm running into the issue where IChannel doesn't expose Category, so the existing code gets confused. Would the answer be to check if the channel has a Category attribute? We'd be able to implement the ChannelThread specific code in an `else` after the fact, but it'd be a lot of conditionals.

I'm also thinking about the order in which we want to download things. Currently with the whole huge block of code for ChannelThread copy-pasted below, it'll begin downloading threads immediately after all channels have finished downloading. Would it be better to handle this in the command rather than the exporting logic, so it can pass in any list of channels it wants? This could also mean it'd be able to download a single channel and then the threads belonging to that channel before moving onto the next channel, which seems like more expected behavior.

Reply:
> I tried this, but I'm running into the issue where IChannel doesn't expose Category, so the existing code gets confused. Would the answer be to check if the channel has a Category attribute? We'd be able to implement the ChannelThread specific code in an else after the fact, but it'd be a lot of conditionals.

I think the solution would be to rework the `Category` property into something like `string ParentName` which would contain the category name for normal channels and the parent channel name for threads. And also add `ParentId` to both channels and threads.

> I'm also thinking about the order in which we want to download things. Currently with the whole huge block of code for ChannelThread copy-pasted below, it'll begin downloading threads immediately after all channels have finished downloading.

Ideally, you should start all downloads as soon as possible. They are then parallelized and downloaded simultaneously (to a degree which the user allows). I don't think the order is that essential beyond that.

> Would it be better to handle this in the command rather than the exporting logic, so it can pass in any list of channels it wants? This could also mean it'd be able to download a single channel and then the threads belonging to that channel before moving onto the next channel, which seems like more expected behavior.

Not entirely sure that I understood this one. I think if the user wants to download a channel and all its threads, the underlying code will still pull the corresponding `IChannel` object for each of them and make multiple export requests (since there's gonna be multiple output file).

Reply:
That makes sense. There's a couple other things that Channel has that ChannelThread does not, like Topic and Position, but that shouldn't be too tricky to work around. I just pushed a commit (caa6aee) that starts this refactor.
To address the second point, as you said it probably just makes sense to treat every IChannel object the same way in `ExportRequest` and `ExportCommandBase`, which I'd imagine would help simplify things as well.

Reply:
Right. For those things, I think the thread implementation of `IChannel` could just provide reasonable defaults, such as `null` for `Topic` and `0` for `Position`.